dspace-spring-rest/src/main/java/org/dspace/app/rest/ItemOwningCollectionUpdateRestController.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.UUID;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.dspace.app.rest.converter.CollectionConverter;
+import org.dspace.app.rest.model.CollectionRest;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.authorize.service.AuthorizeService;
+import org.dspace.content.Collection;
+import org.dspace.content.Item;
+import org.dspace.content.service.CollectionService;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Constants;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.security.access.prepost.PostAuthorize;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/core/items/" +
+        "{itemUuid:[0-9a-fxA-FX]{8}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{12" +
+        "}}/owningCollection")
+public class ItemOwningCollectionUpdateRestController {
+
+    @Autowired
+    ItemService itemService;
+
+    @Autowired
+    CollectionService collectionService;
+
+    @Autowired
+    AuthorizeService authorizeService;
+
+    @Autowired
+    CollectionConverter converter;
+
+    @RequestMapping(method = RequestMethod.PUT)
+    @PreAuthorize("hasPermission(#itemUuid, 'ITEM','WRITE') && hasPermission(#targetUuid,'COLLECTION','ADD')")
+    @PostAuthorize("returnObject != null")
+    public CollectionRest move(@PathVariable UUID itemUuid, HttpServletResponse response,
+                               HttpServletRequest request, @RequestParam(name = "collection") UUID targetUuid)
```

Comment:
This doesn't look to align with the REST Contract (see https://github.com/DSpace/Rest7Contract/pull/34).  The REST Contract stats that the `PUT` request should pass in the Collection's UUID in the *body* of the request (as a `text/uri-list`).   This method looks to be expecting the Collection UUID as a request parameter (`?collection=[:uuid]`)

Reply:
This has been resolved